zarr/tests/test_creation.py

```diff
@@ -429,6 +429,18 @@ def test_create_in_dict(zarr_version, at_root):
     assert isinstance(a.store, expected_store_type)
 
 
+@pytest.mark.skipif(have_fsspec is False, reason="needs fsspec")
```

Comment:
Is fsspec needed to reproduce this bug? In my reading of the code, it it should only affect stores that have a `mode` argument in their constructor. That is:

- `FSStore`
- `ZipStore`
- `DBMStore`

The fact that not all stores accept the `mode` keyword it, to me, a Zarr code smell.

Reply:
@rabernat: would you hold up this PR for using `fsspec` for the regression test?

Also: would you propose an issue to say, "all stores should have a `mode` argument"?